File: java/ArithmeticProg.java
import java.util.Arrays;

public class ArithmeticProg {
    public static void main(String[] args) {
        int [] arr = {1,2,3};
        System.out.println(isInAP(arr));
        int [] arr1 ={19,28,10};
        System.out.println(isInAP(arr1));
    }

    public static boolean isInAP(int[] arr) {
        boolean flag = false;
        Arrays.sort(arr);

        int cd = arr[1] - arr[0];
        for(int i =0; i<arr.length-1; i++){
            if(cd == arr[i+1] - arr[i]){
                flag = true;
            }
            else{
                flag = false;
                return false;
            }
        }    
        return flag;
    }
}

File: js/arithmeticProg.js
function isInAP(array) {
    var flag = false;

    // Sort the array numerically
    array.sort((a, b) => a - b);

    var cd = array[1] - array[0];
    console.log("cd: " + cd);

    for (let i = 0; i < array.length - 1; i++) {
        if (cd === (array[i + 1] - array[i])) {
            console.log(array[i + 1] - array[i]);
            flag = true;
        } else {
            flag = false;
            return flag;
        }
    }
    return true;
}

console.log(isInAP([1, 2, 3]));    // true
console.log(isInAP([1, 2, 8]));    // false
console.log(isInAP([19, 28, 10]));  // true
